src/components/timeline/TrackThread.js

```diff
@@ -145,26 +154,41 @@ class TimelineTrackThread extends PureComponent<Props> {
             // JavaScript and props instead.
             className={`
               timelineTrackThreadIntervalMarkerOverview
-              timelineTrackThreadIntervalMarkerOverviewThread${thread.name}
+              timelineTrackThreadIntervalMarkerOverviewThread${
+                filteredThread.name
+              }
             `}
             rangeStart={rangeStart}
             rangeEnd={rangeEnd}
             threadIndex={threadIndex}
             onSelect={this._onIntervalMarkerSelect}
           />
         ) : null}
-        <StackGraph
-          interval={interval}
-          thread={thread}
-          rangeStart={rangeStart}
-          rangeEnd={rangeEnd}
-          callNodeInfo={callNodeInfo}
-          selectedCallNodeIndex={selectedCallNodeIndex}
-          categories={categories}
-          onSampleClick={this._onSampleClick}
-        />
+        {timelineType === 'category' ? (
+          <ThreadActivityGraph
+            className="threadActivityGraph"
+            interval={interval}
+            fullThread={fullThread}
+            rangeStart={rangeStart}
+            rangeEnd={rangeEnd}
+            onSampleClick={this._onSampleClick}
+            categories={categories}
+          />
+        ) : (
+          <ThreadStackGraph
+            className="threadStackGraph"
+            interval={interval}
+            thread={fullThread}
```

Comment:
In the past, we were using the filteredThread here. I'm leaving it up to you to decide what you think would be more appropriate. (Before this PR, JS only mode was able to remove samples from the graph and affect the displayed stack heights. Not sure if that's useful behavior.)